### PR TITLE
feat: add `--only-templates` flag

### PR DESCRIPTION
**Pull request by sadeghbarati**
For instance, check this PR  in TanStack, https://github.com/TanStack/query/pull/7861

The problem is when repository has too many packages the amount of scroll on the PR page will increase, adding another detail to the pkg.pr.new links conditional with a flag like `--collapsed-links` will help resolve this case

### CODE REVIEW

File: packages/backend/server/utils/markdown.ts
Comment (sadeghbarati) [inline packages/backend/server/utils/markdown.ts:87]: @Aslemammad Hi Mohammad please suggest a heading with `ctrl + g`

File: packages/cli/index.ts
Comment (sadeghbarati) [inline packages/cli/index.ts:69]: Here too 🙏

**Comment by Aslemammad** [comment]
Can we have `onlyTemplates` option that only shows the user defined options? Since that's what Seb suggested. 

instead of the --collapsedLinks option.

**Comment by slorber** [comment]
What I'd like is something like this:

> This PR has been continuously released. You can run its code in one of our templates:
> 
> - [template-js](https://example.com)
> - [template-ts](https://example.com)
> 
> Alternatively you can try the PR code directly into your project:
> 
> <details>
>   <summary>Dependencies</summary>
> 
>   npm add https://pkg.pr.new/@docusaurus/core@7861
>   npm add https://pkg.pr.new/@docusaurus/utils@7861
>   npm add https://pkg.pr.new/@docusaurus/theme-common@7861
> 
> </details>


---

Maybe you can consider the comment has 2 sections.

And let me activate sections and order them? For example `--section templates --section dependencies`

And let me pass a custom section intro message for both.


---

Full example CLI options:


```bash
--section templates
--templates-intro 'This PR has been continuously released. You can run its code in one of our templates:'
--section dependencies
--dependencies-intro 'Alternatively you can try the PR code directly into your project:'
```

**Comment by Aslemammad** [comment]
@slorber For now, let's only show the templates for the option `--onlyTemplates`

For further customization, as #173 suggested, we'd have a json output that would allow maintainers have more control over the output! 

wdyt?

**Comment by slorber** [comment]
Fine with a simple option 👍

**Review by Aslemammad — APPROVED** [review]

**Comment by slorber** [comment]
let me know once it's on npm

By the way what's the final comment with this new option? 

Is the comment template part of this repo? I may want to fork it and customize for our own needs eventually.

**Comment by slorber** [comment]
> Hmm, could you give me more context?

Hmmm not sure what to add: we agreed to keep this option simple and not have the ability to provide custom labels like my former suggestion. Can I implement it myself by forking/patching this package, or is the final message being created/posted by some StackBlitz internal backend?

**Comment by Aslemammad** [comment]
> Can I implement custom labels myself by forking/patching this package, or is the final message being created/posted by some StackBlitz internal backend?

Oh yea, with the #183, now you can turn comments off, get the json output and then create your own message! 😄

We will release soon.

**Comment by slorber** [comment]
@Aslemammad that probably works but looks more complicated, do you have a full github action example flow doing so? 

As far as I remember it's a bit annoying in terms of security to have your action post a GitHub comment and build the code.

**Comment by Aslemammad** [comment]
> As far as I remember it's a bit annoying in terms of security to have your action post a GitHub comment and build the code.

Yea, haven't looked into that, but for sure, it might have its own trade offs. But I guess that's only way we can make it extensible and let others custmize on top of it, without building many abstractions in code, which is a bit challenging.

@pngwn might want to share thoughts on this?

**Comment by pngwn** [comment]
Yes we do this a lot over on gradio. It is indeed a little more annoying but relatively straightforward. 

I'll post an example flow shortly.

**Comment by pngwn** [comment]
Fwiw I do think the JSON approach is preferable to additional formatting params. This allow the most flexibility, and can even allow you to add the info to an existing CI reply rather than every service generating additional replies (which also gets very noisy).

**Comment by slorber** [comment]
Yes JSON is flexible, now if you tell me how to wire it in my CI properly and I just have to copy/paste the complex actions files, that works for me 😄

**Comment by pngwn** [comment]
Just waiting for a release and I'll post it (the old pr link is dead now). Has this been released?

**Comment by pngwn** [comment]
I'll also make a PR to the docs / readme when it is out.

**Comment by Aslemammad** [comment]
Hey not yet, we want to batch it with few other goodies and then!

Sorry for that.

**Comment by pngwn** [comment]
No worries!

**Pull request merged**
